src/components/profile/UserProgress/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { Typography } from '@/components/common';
+import { Variant } from '@/components/common/Typography';
+import { type PublicProfile } from '@/lib/types/apiResponses';
+import { getLevel, getUserRank } from '@/lib/utils';
+import { useEffect, useState } from 'react';
+import styles from './style.module.scss';
+
+export interface UserProgressProps {
+  user: PublicProfile;
+  points: number;
+  isSignedInUser: boolean;
+  levelTextVariant?: Variant;
+  levelDescriptionVariant?: Variant;
+}
+
+export const UserProgress = ({
+  user,
+  points,
+  isSignedInUser,
+  levelTextVariant,
+  levelDescriptionVariant,
+}: UserProgressProps) => {
+  const { firstName } = user;
+  // animate the progress bar
+  const [progress, setProgress] = useState<Number>(0);
+  useEffect(() => setProgress(points % 100), [points]);
```

Comment:
This could probably also be done with a CSS animation with a CSS variable, which would avoid needing a re-render

Reply:
Not an expert on this but I think one nice thing from this is that the points animation will jump by 10 right after a successful checkin

Reply:
you can use a CSS transition. something like

```scss
@keyframes points {
  from {
    width: 0;
  }
  to {
    width: var(--progress);
  }
}
.inner {
  animation: points .5s;
  width: var(--progress);
  transition: width .2s;
}
```

then you can just set the `--progress` style to a percentage